### ISSUE TITLE
Add Rust SDK

### ISSUE THREAD

**Issue by dborovcanin** [Mainflux/mainflux-lite#2545]
### Is your feature request related to a problem? Please describe.

No. I just want to be able to use API Magistrala with Rust client.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

The title is self-explanatory.

### Indicate the importance of this feature to you.

Must-have

**Comment by dborovcanin** [comment]
@dborovcanin Needs to re-review.

**Comment by dborovcanin** [comment]
@dorcaslitunya Is this ready for re-review?

**Comment by dorcaslitunya** [comment]
Yes...it can be re-reviewed. Just a few  questions on design considerations:
1. Would it be more useful to have the models and the apis in one folder? Or is the current implementation okay?
2. Following how examples have been given in the Go SDK, they have been written down in the interface file and laid out there. I currently just have an example.rs file. Should I add more examples here or document them separately?

**Comment by dorcaslitunya** [comment]
Closed but needs further testing on supermq, add test files also.
